### ISSUE TITLE
Basecurve → Preserve colors as new default

### ISSUE THREAD

**Issue by timur-davletshin** [darktable-org/darktable#3677]
dt 3.0 is using preserve color = luminance as new default. This gives results different from in camera JPEGs. Which is a good start for new edits especially from new users — they expect results close to in camera conversion.

I propose to revert old defaults preserve color = none.

In camera JPEG:
![In camera JPEG](https://user-images.githubusercontent.com/58066490/70371774-80c38d00-18e8-11ea-9a0c-82d4d7cb058a.jpg)

Old default (no preserve colors):
![Basecurve (old default)](https://user-images.githubusercontent.com/58066490/70371815-e0ba3380-18e8-11ea-89bc-10bc0543c73b.jpg)

New default (preserve color = luminance):
![Basecurve (new default)](https://user-images.githubusercontent.com/58066490/70371831-08a99700-18e9-11ea-97cf-fae73104c26b.jpg)

Raw image:
https://drive.google.com/open?id=1vY9rmpcNYz5qTSHzwcwETgAKeOut24t1

Difference is even stronger on sunset or bright sky scenes when color shift is very visible.

**Comment by TurboGit** [comment]
I suppose that it really depends on the actual Jpeg and so camera and Basecurve for it in dt. Some of the dt basecure are close to Jpeg, some are not. So I'm really not sure about this.

Now, if in your case you have better default with no color preservation why not just create an auto preset that will replace the default one?

**Comment by timur-davletshin** [comment]
@TurboGit this exactly what I do. But it seems to be the case with all raw file which have builtin JPEGs for comparison.

**Comment by TurboGit** [comment]
Ok, I have nothing against reverting to None as default.

In fact advanced users will probably be disabling basecure and so having a default that is consistent with 2.6 makes sense. I'll propose a PR to be discussed.

**Comment by TurboGit** [comment]
See #3678 

Note that personally I find the result with luminance as color preservation more pleasant, maybe not closer to the in-camera jpeg, but more pleasant.

**Comment by timur-davletshin** [comment]
@TurboGit as I said in mailing list already I agree that it depends on perception and in some cases it provides better results. But first time users tend to compare dt results with in camera JPEG and report differences as bugs.

**Comment by rawfiner** [comment]
I would prefer keeping the color preservation activated for the default. Maybe though setting it to luminance is not the best? @timur-davletshin have you tried the other modes (in particular the power norm), to see if one of them gives you something closer to jpgs?

**Comment by timur-davletshin** [comment]
Sure, I tried and I find previous default choice to be more universal.

**Comment by TurboGit** [comment]
I agree with rawfiner. I have closed my PR we really do expect the new default to be more natural. Not what we used to have in 2.6, right but more natural and people will certainly get used to this.

**Comment by timur-davletshin** [comment]
@TurboGit just to illustrate how highlights behave:

![Untitled](https://user-images.githubusercontent.com/58066490/70379243-57801c80-193b-11ea-828f-f71e17ef670f.png)

There is no real sensor saturation in that image. Left side is old default and it handles highlights correctly, right one is new. Those marked red have slight magenta cast compared to left and in camera JPEG. I find this dramatic right shift very bad in similar situations. I observe it in all similar images with bright sky.

**Comment by timur-davletshin** [comment]
Question: Is it good to make huge red blod results from any high dynamic range (yet properly exposed) picture?

**Comment by angryziber** [comment]
The new default produces really desaturated pictures of snow (my trip to Svalbard has lots of it).
The pictures look really bleak :-( I was looking for the reason and finally found that the new feature has appeared in basecurve that does this

**Comment by timur-davletshin** [comment]
@angryziber I don't like it either, but developers see it as new default. It desaturates midtones, causes highlights overexposure but in some situation in brings good results. It gives better color rendition for some sky pictures. IMO making it default is a mistake.

**Comment by angryziber** [comment]
Haven't seen any good results so far, but a plenty of bad ones...
From the discussion it seems that developers are not using the Basecurve module themselves, that's probably the reason they are not seeing the problem.
@TurboGit is there a chance to reconsider?

**Comment by TurboGit** [comment]
@angryziber : This has been debated again and again. A conscious decision has been made among the developers. Can we move on?

If you prefer the old behavior fine, create an auto-preset and you'll never see again the new default.

**Comment by angryziber** [comment]
I did it, but new users...

**Comment by timur-davletshin** [comment]
@TurboGit no offense, but where is this discussion everybody refers to? @rawfiner failed to provide me with link. I'd like to see pictures with better results. IMO, this new mode just makes ugly results for the whole bunch of pretty normal shots.

**Comment by rawfiner** [comment]
@timur-davletshin I was refering to discussions on your different issues.
The thing is, disabling color preservation will give you results broken in a way that users (especially new users) won't be able to easily determine and fix (why do my colors looks odd?). I prefer having sometimes (in my opinion it happens very rarely though) ugly results, that I can see and fix easily. All the point here is to give user the good habit of preserving their colors.

**Comment by TurboGit** [comment]
The discussion is here and in some other closed issues. I have also discussed that in French in an IRC like channel with some devs.

@timur-davletshin : In your last example I think having the highlight clipped is ok, the picture is bright (too bright as illustrated with the left picture), just play with exposure and you're set with better color preserved. that's a trade off.

Maybe that's related to your camera but I did test again for some of my NEF, all pictures with the basecurve set to luminance are better color wise. Now, please understand that this will be my last message on this issue.

**Comment by timur-davletshin** [comment]
@rawfiner that's just not true. Tried snow — it's just utter failure out of the box. Another example — https://discuss.pixls.us/t/play-raw-sacre-coeur-backlight-challenge/15124 — you may like blue color of that halo, but it's not the way I saw it. It was not blue like new default renders it. Not saying that no one did color checker test before making this decision.

**Comment by timur-davletshin** [comment]
> Maybe that's related to your camera

To avoid this kind of comment I specifically provided samples from different cameras.

**Comment by rawfiner** [comment]
Tried snow too, on several very different pictures, and it renders well with preserve colors activated.
On your other example, the look is very similar, and you just have to desaturate your highlights.
All saturation-related "changes" comparing to old behavior are expected, as the old behavior changed saturation in a non-uniform way across the tones. You can use color balance to boost saturation or reduce it, and parametric masks if needed.

This will also be my last message on this issue and on #3693

**Comment by chrisk314** [comment]
Wow I just got really thrown for a loop by this. I'm shooting exclusively with fuji xt2 and use the "fujilike" base curve by default. Previously with my setup colour saturation looked very natural and close to in camera rendering. At some point in a new version the basecurve changed to preserve - luminance by default. It took me a while to track down what had changed. Images look very muddy and flat now with the default. In particular greens and reds look completely lifeless. Part of this of course is down to perception. I accept the rationale of the devs and the fix is an easy one - just make a new default basecurve. However I did just want to add my voice and say that for the fujilike basecurve the default result to my eyes now look pretty bad and far from the rendering of jpeg in camera on the xt2.

**Comment by irafaz** [comment]
I strongly dislike the new luminance-setting as default as well. I think @timur-davletshin is right when he says a lot of people (including me) expects the old behavior of the basecurve, where the RAW image looked a lot like the JPEG. And I don't care if this is "wrong" color-wise. This is the expected behavior in a lot of photo processing software, including Darktable.

Wouldn't this be quite easy to implement as a general setting for Darktable? Something like a "preserve colors"-setting? If you're looking to please the advanced users here, I don't think they would mind turning on a setting to "preserve colors". But I _really_ think beginners or hobby-photographers don't want this behavior as the default.

I have created new basecurves for myself which auto-applies to my RAWs, but I can never ever recommend Darktable to someone else if they would have to go through those steps just to get the images to look "normal".

Please do it the other way around, and/or implement a setting where advanced users can change the default behavior. As you have said yourselves, the advanced users might not even use basecurves...

**Comment by aurelienpierre** [comment]
> This is the expected behavior in a lot of photo processing software, including Darktable.

Expected by whom ? Do you have a poll ? If you like your camera JPEG, you don't need darktable.

Create a preset with chroma preservation off, and set it to auto-apply for new raw.

> Previously with my setup colour saturation looked very natural and close to in camera rendering.

Except it wasn't natural at all.

**Comment by irafaz** [comment]
> > This is the expected behavior in a lot of photo processing software, including Darktable.
> 
> Expected by whom ? Do you have a poll ? If you like your camera JPEG, you don't need darktable.
> 
> Create a preset with chroma preservation off, and set it to auto-apply for new raw.
> 
> > Previously with my setup colour saturation looked very natural and close to in camera rendering.
> 
> Except it wasn't natural at all.

Let me rephrase; That's how I have worked in other photo processing software and thus I expect it to work that way.

I am quite happy with the JPEGs from my cameras. But the RAW images allow me to manipulate them further and bringing back details and recovering mistakes afterwards in a way I can not do with my JPEGs. Hence, the base curve that almost represents the camera JPEG was a great beginning point for my editing workflow. And I suspect (without statistics and only by the knowledge about how other photographers I know works) that most other peoples workflows look(ed) the same. Except for the most advanced photo retouchers, I continue to assume.

Even if I know how create new presets and auto-apply them, I really can't expect new users to understand that. And I really can't expect them to take their time to look up why their pictures "look weird" in comparison to how they have looked before here on GitHub. I really can't recommend Darktable and its workflow to any hobby photographer I know that currently uses other software.

Wouldn't it be easier to keep the standard settings as easy as possible for new users, beginners and intermediate photographers, and then let them explore Darktable and its functions and let them explore wider colors and their own RAW-conversions and such?

**Comment by TurboGit** [comment]
> I strongly dislike the new luminance-setting as default as well.

I strongly like the new luminance-setting as default as well.

Have we made progress ?

Are we going to have this discussion again and again ?

Can't you create a preset and move on ?

We have two group, there is no strong winner. So please let close this discussion except if there is new string mathematical evidence about that such or such option is best in 90% of the case.

**Comment by irafaz** [comment]
> > I strongly dislike the new luminance-setting as default as well.
> 
> I strongly like the new luminance-setting as default as well.
> 
> Have we made progress ?
> 
> Are we going to have this discussion again and again ?
> 
> Can't you create a preset and move on ?
> 
> We have two group, there is no strong winner. So please let close this discussion except if there is new string mathematical evidence about that such or such option is best in 90% of the case.

Then can we start a poll where most of the Darktable-users will see it and actually vote?

Would you rather;
1. That the base curve mimics the cameras JPEG tones, or
2. That the base curve give you a wider color starting point for your processing?

I still think option 1 is easier to grasp for novices and thus I think it would be easier to use as default for Darktable. I think option 2 sounds like a great option/addition for advanced photographer in regards to both hardware and software. I understand that they would benefit a lot from it. But I also think they are the ones that would be capable of setting up that workflow for themselves. The novices (like my photography-friends) would not. Hence, I propose that it would be an easy setting to switch to in settings for the advanced user.

What do you think of that idea?

**Comment by chrisk314** [comment]
Let me just say that I love Darktable and I'm very grateful for all the hard work put in by the devs. When I have some free bandwidth I would love to contribute in some way. However, as others have mentioned this is quite off putting for beginners. I'm a software engineer (as probably 95% of people looking here are) so for me looking at a GitHub issue is not a problem but the idea that novice users should have to come here to figure out why their images "don't look how they expect" is comical and naive.

> We have two group, there is no strong winner. So please let close this discussion except if there is new string mathematical evidence about that such or such option is best in 90% of the case.

If you want strong statistical evidence then you would need to run a poll. Without doing so no one can say what the majority of Darktable users would prefer - and of course this metric would give an indication of what potential Darktable users would prefer as well.

**Comment by lord-carlos** [comment]
Oh my.
I switched from 2.6 to latest and could not figoure out why I could not make any person look human.
I gave up and shot JPEG 😐

I personally like it more if a program makes it easy for beginners, but gives flexibility for advanced users.

**Comment by TurboGit** [comment]
> I personally like it more if a program makes it easy for beginners, but gives flexibility for advanced users.

Or learn using dt which is not that difficult, you have plenty of tutorials around. Last but not least darktable does not target beginers but pro or semi-pro photographers wanting to get the best of the RAW file.
